src/pages/TracksPage/store/index.ts

```diff
@@ -8,23 +8,32 @@ import { TracksPageState } from './types';
 
 export const tracksPageSlice: StateCreator<TracksPageState> = (set, get) => ({
   userTracks: [],
+  isFavouriteTracksLoading: false,
   totalTracks: 0,
   isUserTracksLoading: true,
   isQueryTracksLoading: false,
   currentPage: 1,
+  setIsFavouriteTracksLoading: (state: boolean) =>
+    set({ isFavouriteTracksLoading: state }),
   setCurrentPage: (number: number) => set({ currentPage: number }),
   setIsUserTracksLoading: (state) => set({ isUserTracksLoading: state }),
   setIsQueryTracksLoading: (state) => set({ isQueryTracksLoading: state }),
+  clearUserTracks: () => set({ userTracks: [], currentPage: 1 }),
   getTracksByUser: async (
     userId: string,
     page: number,
-    filterOptions = { query: '', sortType: 'updatedAt', order: 'desc' },
+    filterOptions = {
+      query: '',
+      sortType: 'updatedAt',
+      order: 'desc',
+      isFavourite: false,
+    },
   ) => {
-    const { query, sortType, order } = filterOptions;
+    const { query, sortType, order, isFavourite } = filterOptions;
```

Comment:
Тут также присваивай значения по умолчанию
```suggestion
    const { query = '', sortType, order, isFavourite } = filterOptions;
```

Привел пример с query